src/python/zquantum/qcbm/cost_function.py

```diff
@@ -84,13 +49,25 @@ def _evaluate(self, parameters: np.ndarray) -> (float, BitstringDistribution):
             (float): cost function value for given parameters
             zquantum.core.bitstring_distribution.BitstringDistribution: distribution obtained
         """
-        circuit = self.ansatz.get_executable_circuit(parameters)
-        distribution = self.backend.get_bitstring_distribution(circuit)
+        circuit = ansatz.get_executable_circuit(parameters)
+        distribution = backend.get_bitstring_distribution(circuit)
         value = evaluate_distribution_distance(
-            self.target_bitstring_distribution,
+            target_bitstring_distribution,
             distribution,
-            self.distance_measure,
-            distance_measure_parameters=self.distance_measure_parameters,
+            distance_measure,
+            distance_measure_parameters=distance_measure_parameters,
+        )
+
+        if store_artifact:
+            store_artifact("bitstring_distribution", distribution)
+
+        return ValueEstimate(value)
```

Comment:
It's not that important but note that currently, you don't need ValueEstimate, as without precision it is interchangeable with float :slightly_smiling_face: Of course, in such a case you would need to adjust typing in the signature.

Reply:
Good point - leaving it for now to remain consistent with `AnsatzBasedCostFunction`, but we should revisit this laster